src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GeneralApplicationDirectionsService.java

```diff
@@ -0,0 +1,184 @@
+            GENERAL_APPLICATION_DIRECTIONS_HEARING_TIME_ESTIMATE,
+            GENERAL_APPLICATION_DIRECTIONS_HEARING_REGION,
+            GENERAL_APPLICATION_DIRECTIONS_MIDLANDS_FRC,
+            GENERAL_APPLICATION_DIRECTIONS_LONDON_FRC,
+            GENERAL_APPLICATION_DIRECTIONS_NORTHWEST_FRC,
+            GENERAL_APPLICATION_DIRECTIONS_NORTHEAST_FRC,
+            GENERAL_APPLICATION_DIRECTIONS_SOUTHEAST_FRC,
+            GENERAL_APPLICATION_DIRECTIONS_WALES_FRC,
+            GENERAL_APPLICATION_DIRECTIONS_NOTTINGHAM_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_CFC_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_BIRMINGHAM_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_LIVERPOOL_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_MANCHESTER_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_CLEVELAND_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_NWYORKSHIRE_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_HUMBER_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_KENTSURREY_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_NEWPORT_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_SWANSEA_COURT,
+            GENERAL_APPLICATION_DIRECTIONS_ADDITIONAL_INFORMATION,
+            GENERAL_APPLICATION_DIRECTIONS_RECITALS,
+            GENERAL_APPLICATION_DIRECTIONS_JUDGE_TYPE,
+            GENERAL_APPLICATION_DIRECTIONS_JUDGE_NAME,
+            GENERAL_APPLICATION_DIRECTIONS_COURT_ORDER_DATE,
+            GENERAL_APPLICATION_DIRECTIONS_TEXT_FROM_JUDGE
+        ).forEach(generalApplicationDirectionCcdField -> caseData.remove(generalApplicationDirectionCcdField));
+    }
+
+    public void submitGeneralApplicationDirections(CaseDetails caseDetails, String authorisationToken) {
+        CaseDocument document = caseDetails.getData().get(GENERAL_APPLICATION_DIRECTIONS_HEARING_REQUIRED).equals(YES_VALUE)
+            ? prepareHearingRequiredNoticeDocument(caseDetails, authorisationToken)
+            : prepareGeneralApplicationDirectionsOrderDocument(caseDetails, authorisationToken);
+
+        printDocumentPackAndSendToApplicantAndRespondent(caseDetails, authorisationToken, document);
+        updateGeneralApplicationLatestDocument(caseDetails, document);
+        resetStateToGeneralApplicationPrestate(caseDetails);
+    }
+
+    private void printDocumentPackAndSendToApplicantAndRespondent(CaseDetails caseDetails, String authorisationToken, CaseDocument document) {
+        List<BulkPrintDocument> documents = asList(documentHelper.getCaseDocumentAsBulkPrintDocument(document));
+        bulkPrintService.printApplicantDocuments(caseDetails, authorisationToken, documents);
+        bulkPrintService.printRespondentDocuments(caseDetails, authorisationToken, documents);
+    }
+
+    private void updateGeneralApplicationLatestDocument(CaseDetails caseDetails, CaseDocument document) {
+        Map<String, Object> caseData = caseDetails.getData();
+        caseData.put(GENERAL_APPLICATION_DOCUMENT_LATEST, document);
```

Comment:
I don't think you should be updating this as this will no longer be the general application document but the notice document, I don't think that is correct

Reply:
Good point, removed that bit and added other documents to print required in AC.